zarr/storage.py

```diff
@@ -1604,6 +1605,12 @@ class NestedDirectoryStore(DirectoryStore):
     special handling for chunk keys so that chunk files for multidimensional
     arrays are stored in a nested directory tree.
 
+    .. deprecated:: 2.18.0
+            NestedDirectoryStore will be removed in Zarr-Python 3.0 where controlling
+            the chunk key encoding will be supported as part of the array metadata. See
+            `GH1274 <https://github.com/zarr-developers/zarr-python/issues/1274>`_
+            for more information.
```

Comment:
As far as I'm concerned, this is the only controversial thing in this PR.

Reply:
The "Prefixed" Stores are a code smell since they define storage parameters in python-land only rather than in the zarr metadata. I support this change :+1: